Building borked on macOS (python issues, possibly homebrew)

### mpv version and platform

mpv master
macOS 10.13.3

### Reproduction steps

Just build mpv master on a system with up-to-date homebrew install with `./bootstrap.py && ./waf configure` (with options) `&& ./waf build`, then try to make a bundle with the TOOLS/osxbundle.py script.

### Expected behavior

Mpv is built regularly and an app bundle is made.

### Actual behavior

Build exits at linking phase with

> Undefined symbols for architecture x86_64:
>   "_av_codec_iterate", referenced from:
>       _add_codecs in av_common.c.20.o
>       _encode_lavc_showhelp in encode_lavc.c.20.o
>       _add_all_hwdec_methods in vd_lavc.c.20.o
>   "_av_demuxer_iterate", referenced from:
>       _mp_get_lavf_demuxers in av_common.c.20.o
>       _demux_open_lavf in demux_lavf.c.20.o
>   "_av_frame_new_side_data_from_buf", referenced from:
>       _mp_image_to_av_frame in mp_image.c.20.o
>   "_av_mallocz_array", referenced from:
>       _mp_aframe_pool_allocate in aframe.c.20.o
>   "_av_muxer_iterate", referenced from:
>       _encode_lavc_showhelp in encode_lavc.c.20.o
>   "_avcodec_get_hw_config", referenced from:
>       _add_all_hwdec_methods in vd_lavc.c.20.o
>   "_avcodec_get_hw_frames_parameters", referenced from:
>       _get_format_hwdec in vd_lavc.c.20.o
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)]

If I manage to compile mpv nonetheless (see below), running the osxbundle scripts fails with the following error

> $ TOOLS/osxbundle.py build/mpv
> Creating Mac OS X application bundle (version: 0.28.0-UNKNOWN)...
> > copying bundle skeleton
> > copying binary
> > create bundle symlink
> > generating Info.plist
> > bundling dependencies
> Traceback (most recent call last):
>   File "TOOLS/dylib-unhell.py", line 94, in <module>
>     main()
>   File "TOOLS/dylib-unhell.py", line 89, in main
>     libs = libraries(binary)
>   File "TOOLS/dylib-unhell.py", line 38, in libraries
>     result[objfile] = set(libs_list)
>   File "TOOLS/dylib-unhell.py", line 15, in is_user_lib
>     return not sys_re.match(libname) and \
> TypeError: cannot use a string pattern on a bytes-like object
> done.

I may be wrong, but I _strongly_ suspect this breakage is due to the fact that the homebrew package manager recently migrated to python3 as default, so now `python` is linked to homebrew's python 3.6.4 instead of the usual system's python 2.7.10: if I temporarily unlink the brew package the builds succeed ad usual. Also, using the `python2.7` command directly "solves" the waf issue, but not the osxbundle.py one (maybe because it automatically picks the default python version with env, unlike waf?).

--- a/TOOLS/dylib-unhell.py
+++ b/TOOLS/dylib-unhell.py
@@ -24,8 +24,8 @@ def is_user_lib(objfile, libname):
 
 def otool(objfile):
     command = "otool -L %s | grep -e '\t' | awk '{ print $1 }'" % objfile
-    output  = subprocess.check_output(command, shell = True)
-    return filter(partial(is_user_lib, objfile), output.split())
+    output  = subprocess.check_output(command, shell = True, universal_newlines=True)
+    return set(filter(partial(is_user_lib, objfile), output.split()))
 
 def install_name_tool_change(old, new, objfile):
     subprocess.call(["install_name_tool", "-change", old, new, objfile])
@@ -35,7 +35,7 @@ def install_name_tool_id(name, objfile):
 
 def libraries(objfile, result = dict()):
     libs_list       = otool(objfile)
-    result[objfile] = set(libs_list)
+    result[objfile] = libs_list
 
     for lib in libs_list:
         if lib not in result:
@@ -80,15 +80,17 @@ def process_libraries(libs_dict, binary, processed = []):
             if p in libs_dict[src]:
                 install_name_tool_change(p, lib_name(p), dst)
 
-    process_libraries(libs_dict, binary, ls)
+    return ls
 
 def main():
     binary = os.path.abspath(sys.argv[1])
     if not os.path.exists(lib_path(binary)):
         os.makedirs(lib_path(binary))
     libs = libraries(binary)
-    print(libs)
-    process_libraries(libs, binary)
+
+    libs_processed = process_libraries(libs, binary)
+    while libs_processed is not None:
+        libs_processed = process_libraries(libs, binary, libs_processed)
 
 if __name__ == "__main__":
     main()

